src/lib.rs

```diff
@@ -192,47 +222,172 @@ impl fmt::Debug for SignatureShare {
     }
 }
 
-/// A secret key.
-#[derive(Clone, PartialEq, Eq, Rand)]
-pub struct SecretKey(Fr);
+/// A secret key; wraps a single prime field element. The field element is
+/// heap allocated to avoid any stack copying that results when passing
+/// `SecretKey`s between stack frames.
+#[derive(PartialEq, Eq)]
+pub struct SecretKey(Box<Fr>);
 
-impl fmt::Debug for SecretKey {
```

Comment:
Instead of removing the `Debug` impl, could you simply change it to display something like `SecretKey(<not shown>)` instead? Not implementing debug means that types which store a SecretKey have to manually impl `Debug` if they need it.

Reply:
Or just `SecretKey(...)`.

Reply:
Yup it should be, I missed adding the `Debug` impl when copying the code over from my `hbbft` branch.

Reply:
Whoops :)